lib/core/include/irods/irods_server_properties.hpp

```diff
@@ -136,18 +148,37 @@ namespace irods
 
         void remove( const std::string& _key );
 
+        /// \brief Get a reference to the underlying json object. Writing
+        /// in particular should be done with caution as there is no guarantee
+        /// that this will be safe.
         nlohmann::json& map() noexcept
         {
             return config_props_;
         }
 
     private:
+        /// @brief Acquire a read lock, that is, a lock that is not exclusive, that is why it returns a shared_lock
+        /// here.
+        std::shared_lock<std::shared_mutex> acquire_read_lock() const
+        {
+            return std::shared_lock<std::shared_mutex>(property_mutex_);
+        }
+
+        /// @brief Acquire an exclusive lock for writing. While as many threads may read the properties object,
+        /// writing to the underlying std::map(or close enough, nlohmann uses red black trees) is not guaranteed to be
+        /// thread safe
+        std::unique_lock<std::shared_mutex> acquire_write_lock() const
+        {
+            return std::unique_lock(property_mutex_);
```

Comment:
Add a comment stating `std::unique_lock` is used instead of `std::scoped_lock` and `std::lock_guard` because it supports move semantics.

While using the other types could work (because of RVO), we don't want to rely on that, therefore `std::unique_lock` is the correct choice.